packages/@aws-cdk/aws-amplify-alpha/lib/asset-deployment-handler/handler.ts

```diff
@@ -49,22 +48,22 @@ export class AmplifyAssetDeploymentHandler extends ResourceHandler {
     }
 
     // Create a pre-signed get URL of the asset so Amplify can retrieve it.
-    const assetUrl = this.s3.getSignedUrl('getObject', {
+    const command = new GetObjectCommand({
       Bucket: this.props.S3BucketName,
       Key: this.props.S3ObjectKey,
     });
+    const assetUrl = await getSignedUrl(this.s3, command);
 
     // Deploy the asset to Amplify.
     const deployment = await this.amplify
       .startDeployment({
         appId: this.props.AppId,
         branchName: this.props.BranchName,
         sourceUrl: assetUrl,
-      })
-      .promise();
+      });
 
     return {
-      AmplifyJobId: deployment.jobSummary.jobId,
+      AmplifyJobId: deployment.jobSummary?.jobId,
```

Comment:
When is `deployment.jobSummary` not defined? And could this possibly be a breaking change?

Reply:
This is just a side effect of better typings in the new versions of the SDKs. My understanding is the same failure cases existed before where the response was possibly undefined but just not typed explicitly and therefore not handled.